.github/workflows/load-db-snaphot.yml

```diff
@@ -0,0 +1,205 @@
+name: Get and Load django db snapshot into environment
+
+on:
+  workflow_dispatch:
+    inputs:
+      instance:
+        description: 'Cluster and namespace to load the snapshot into'
+        default: 'hydroshare-dev\dev-0'
+        required: true
+        type: choice
+        options:
+          - 'hydroshare-beta\default'
+          - 'hydroshare-dev\dev-0'
+          - 'hydroshare-dev\dev-1'
+          - 'hydroshare-dev\dev-2'
+
+      snapshot:
+        description: 'The snapshot to load, default is latest or use the date of the snapshot file e.g 20250819 you want as the current format for the name is e.g backup-production-jjjjmmdd.sql'
+        default: 'latest'
+        required: true
+        type: string
+jobs:
+  deploy:
+    name: Run on ${{ inputs.instance }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: code checkout
+      uses: actions/checkout@v3
+
+    - uses: 'google-github-actions/auth@v2'
+      with:
+        #credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}
+        credentials_json: ${{ secrets.HYDROSHARE_STORAGEREADER_KEY}}
+
+    - name: Setup GKE credentials for dev-
+      if: ${{ contains(inputs.instance, 'dev-') }}
+      uses: google-github-actions/get-gke-credentials@v2
+      with:
+        cluster_name: hydroshare-dev
+        location: us-central1
+
+    - name: Setup GKE credentials for beta
+      if: ${{ contains(inputs.instance, 'beta') }}
+      uses: google-github-actions/get-gke-credentials@v2
+      with:
+        cluster_name: hydroshare-beta
+        location: us-central1
+        
+    - name: Set context for dev-0
+      if: ${{ contains(inputs.instance, 'dev-0') }}
+      run: |
+        echo "setting context to dev-0"
+        kubectl config use-context gke_hydroshare-403701_us-central1_hydroshare-dev 
+        kubectl config set-context --current --namespace=dev-0
+
+    - name: Set context for dev-1
+      if: ${{ contains(inputs.instance, 'dev-1') }}
+      run: |
+        echo "setting context to dev-1"
+        kubectl config use-context gke_hydroshare-403701_us-central1_hydroshare-dev 
+        kubectl config set-context --current --namespace=dev-1
+    
+    - name: Set context for dev-2
+      if: ${{ contains(inputs.instance, 'dev-2') }}
+      run: |
+        echo "setting context to dev-2"
+        kubectl config use-context gke_hydroshare-403701_us-central1_hydroshare-dev 
+        kubectl config set-context --current --namespace=dev-2
+
+    - name: Set context for beta
+      if: ${{ contains(inputs.instance, 'beta') }}
+      run: |
+        echo "setting context to beta"
+        kubectl config use-context gke_hydroshare-403701_us-central1_hydroshare-beta
+        kubectl config set-context --current --namespace=default
```

Comment:
you can avoid these duplicated steps if you extract the cluster name from the inputs.instance.

Here is an example of how that would work:
https://github.com/hydroshare/hydroshare/pull/6096/commits/069e657c24aa876c9330f2b7773c81fd99b3e83f